lib/puppet/type/component.rb

```diff
@@ -49,6 +49,11 @@ def ref
     reference.to_s
   end
 
+  # This allows us to grab the defined type 'type' for optional use in other custom type's autorelation methods
+  def title_class
```

Comment:
The name of this method is odd, and the documentation does not help. It should explain what it returns first and then explain the typical use case (if not apparent from the description). Also (I have not looked), if there already is a way to get `reference` that is API, that should be used instead of adding more convenience methods.